gradle build not working

When you run `gradle jar` or `gradle test` getting the following error:
```
Where:
Build file '/Users/belekayazbekov/Homeworks/Integration/ivt-hw-2023-maddevs/build.gradle' line: 3

* What went wrong:
A problem occurred evaluating root project 'ivt-hw-2023-maddevs'.
> Plugin with id 'maven' not found.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 547ms 
```

This is happening probably because this project was created 4 years ago and uses some outdated version of some tools

--- a/build.gradle
+++ b/build.gradle
@@ -1,10 +1,10 @@
 apply plugin: "idea"
 apply plugin: "java"
-apply plugin: "maven"
+apply plugin: "maven-publish"
 
-task wrapper(type: Wrapper) {
-	gradleVersion = '2.0'
-	distributionUrl = "http://services.gradle.org/distributions/gradle-${gradleVersion}-all.zip"
+wrapper {
+	gradleVersion = '8.1.1'
+	distributionUrl = "https://services.gradle.org/distributions/gradle-${gradleVersion}-all.zip"
 }
 
 sourceSets {
@@ -41,29 +41,30 @@ repositories {
 	mavenCentral()
 	maven {
 		url "http://maven.stardog.com"
+		allowInsecureProtocol = true
 	}
 }
 
 dependencies {
-	compile (group: "com.google.guava", name: "guava", version: "18.0") {
+	implementation (group: "com.google.guava", name: "guava", version: '30.0-jre') {
 		exclude group: "com.google.code.findbugs"
 	}
 
-	compile "commons-beanutils:commons-beanutils:1.9.2"
-	compile "commons-codec:commons-codec:1.10"
+	implementation 'commons-beanutils:commons-beanutils:1.9.4'
+	implementation 'commons-codec:commons-codec:1.13'
 
-	compile "com.complexible.common:cp-openrdf-utils:4.0.2"
-	compile ("com.complexible.common:cp-common-utils:5.0.1") {
+	implementation "com.complexible.common:cp-openrdf-utils:4.0.2"
+	implementation ("com.complexible.common:cp-common-utils:5.0.1") {
 		exclude group: "org.slf4j"
 	}
 
 	// these should be coming from cp-openrdf-utils, but it's missing these
-	compile "org.openrdf.sesame:sesame-rio-ntriples:4.0.0"
+	implementation "org.openrdf.sesame:sesame-rio-ntriples:4.0.0"
 
-	compile "org.slf4j:slf4j-api:1.7.7"
-	compile "org.slf4j:slf4j-jdk14:1.7.7"
+	implementation "org.slf4j:slf4j-api:1.7.7"
+	implementation "org.slf4j:slf4j-jdk14:1.7.7"
 
-	testCompile 'junit:junit:4.11'
+	testImplementation 'junit:junit:4.13.1'
 }
 
 tasks.withType(Test) {

--- a/documentation.md
+++ b/documentation.md
@@ -1,0 +1,34 @@
+# Documentation
+
+## Issue #1: Updating Gradle and Modifying Deprecated Configurations
+
+### Overview
+
+During our efforts to build the project, we encountered several issues that were rooted in the 
+outdated version of Gradle that we were using. It quickly became clear that we needed to 
+update Gradle to a newer version to solve these problems and align ourselves with the current 
+best practices.
+
+Upon updating Gradle, we found that several configurations used in our `build.gradle` file were 
+deprecated or completely removed in the newer versions of Gradle. This required us to revise our 
+`build.gradle` file and update our dependency configurations.
+
+### Main Results
+
+The transition to a newer version of Gradle and the update of the `build.gradle` file was successful.
+We replaced the `compile` configuration with `implementation` and `testCompile` with `testImplementation`
+as per the new Gradle conventions.
+
+By adapting to these new conventions, we have managed to bring our project up to date with the current 
+Gradle practices, enhancing our build performance and dependency management. This update also ensures 
+that our project can leverage the latest features and improvements offered by Gradle.
+
+### Lessons Learnt
+
+This process reiterated the importance of maintaining updated build tools in our development 
+environment. By updating Gradle and aligning our project with the latest conventions, we have 
+improved the efficiency of our builds and learned a great deal about Gradle's dependency management
+and the implications of its different configurations.
+
+Going forward, we have recognized the need to stay updated with the changes and enhancements in our
+build tools to maximize our development efficiency and ensure the robustness of our builds.

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Thu Nov 07 20:19:05 CET 2019
+#Tue Mar 31 12:42:31 EDT 2015
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.1.1-all.zip

